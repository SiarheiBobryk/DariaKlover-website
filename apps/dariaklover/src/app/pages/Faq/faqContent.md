# Часто задаваемые вопросы

1.  _Хочу записаться на консультацию, но нет подходящего места. Что делать?_

    Пожалуйста, свяжитесь со мной через [Telegram](https://t.me/dariaklover), [WhatsApp](https://wa.me/+46738540708) или по [Email](mailto:dariaklover@gmail.com).

2.  _Как подготовиться к сессии?_

    К сессии необходимо подготовить запрос. Тихое и комфортное место, где вас не будут отвлекать. Блокнот для записей и ручку.

3.  _Что такое "запрос"?_

    Запрос - это задача, которую вы хотите решить с коучем в ходе сессии.

4.  _Со скольки лет можно работать с коучем?_

    С 18 лет, когда человек сам за себя отвечает. Уровень ответственности играет важную роль в работе с коучем.

5.  _Сколько нужно сессий, чтобы решить запрос?_

    Все индивидуально. Зависит от глубины запроса, вашей вовлеченности и комфортного для вас темпа решения задачи.
