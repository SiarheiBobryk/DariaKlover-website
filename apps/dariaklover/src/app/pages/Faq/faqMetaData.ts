export interface FaqMetaData {
  title: string;
  heading: string;
  path: string;
}

const faqMetaData: FaqMetaData = {
  title: 'Часто задаваемые вопросы',
  heading: 'Часто задаваемые вопросы',
  path: '/help/faq',
};

export default faqMetaData;
