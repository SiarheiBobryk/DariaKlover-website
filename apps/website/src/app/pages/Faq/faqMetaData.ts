import { PageMetaData } from '@dariaklover/types';

const faqMetaData: PageMetaData = {
  title: 'Часто задаваемые вопросы',
  heading: 'Часто задаваемые вопросы',
  path: '/help/faq',
  breadcrumb: 'FAQ',
};

export default faqMetaData;
