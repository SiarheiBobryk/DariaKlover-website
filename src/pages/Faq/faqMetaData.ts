export interface FaqMetaData {
  title: string;
  heading: string;
  path: string;
}

const howToBookMetaData: FaqMetaData = {
  title: 'Часто задаваемые вопросы',
  heading: 'Часто задаваемые вопросы',
  path: '/faq',
};

export default howToBookMetaData;
