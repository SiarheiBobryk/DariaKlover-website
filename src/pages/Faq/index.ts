export { default } from './Faq';
export { default as faqMetaData } from './faqMetaData';
