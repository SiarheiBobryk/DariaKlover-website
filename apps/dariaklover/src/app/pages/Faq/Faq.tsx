import * as React from 'react';
import { Helmet } from 'react-helmet';

import Box from '@mui/material/Box';
import Typography from '@mui/material/Typography';

import faqMetaData from './faqMetaData';
import Heading from '../../components/Heading';
import LinkEmail from '../../components/LinkEmail';
import LinkSocial from '../../components/LinkSocial';
import Paragraph from '../../components/Paragraph';

export interface FaqData {
  question: string;
  answer: string | React.ReactNode;
  id: string;
}

export const faq: Array<FaqData> = [
  {
    question: 'Хочу записаться на консультацию, но нет подходящего места. Что делать?',
    answer: (
      <>
        Пожалуйста, свяжитесь со мной через <LinkSocial social="telegram" />, <LinkSocial social="whatsapp" /> или по{' '}
        <LinkEmail />.
      </>
    ),
    id: 'no-slots',
  },
  {
    question: 'Как подготовиться к сессии?',
    answer:
      'К сессии необходимо подготовить запрос. Тихое и комфортное место, где вас не будут отвлекать. Блокнот для записей и ручку.',
    id: 'how-to-prepare',
  },
  {
    question: 'Что такое "запрос"?',
    answer: 'Запрос - это задача, которую вы хотите решить с коучем в ходе сессии.',
    id: 'what-is-request',
  },
  {
    question: 'Со скольки лет можно работать с коучем?',
    answer:
      'С 18 лет, когда человек сам за себя отвечает. Уровень ответственности играет важную роль в работе с коучем.',
    id: 'age-restrictions',
  },
  {
    question: 'Сколько нужно сессий, чтобы решить запрос?',
    answer:
      'Все индивидуально. Зависит от глубины запроса, вашей вовлеченности и комфортного для вас темпа решения задачи.',
    id: 'sessions-amount-required',
  },
];

function indexToNumber(index: number): number {
  return index + 1;
}

function Faq() {
  return (
    <>
      <Helmet>
        <title>{faqMetaData.title}</title>
      </Helmet>
      <Heading>{faqMetaData.heading}</Heading>
      {/* Uncomment it when we get a huge list of question */}
      {/* <List>
        {faq.map(function mapFaq(faqData: FaqData, index: number) {
          const { question, id } = faqData;
          return (
            <ListItem key={id}>
              <Link href={`#${id}`}>
                <Typography component="h3" sx={{ fontStyle: 'italic' }}>
                  {indexToNumber(index)}. {question}
                </Typography>
              </Link>
            </ListItem>
          );
        })}
      </List> */}
      {faq.map(function mapFaq(faqData: FaqData, index: number) {
        const { question, answer, id } = faqData;
        return (
          <Box id={id} component="section" key={id} sx={{ marginY: 2 }}>
            <Typography component="h2" sx={{ fontStyle: 'italic' }}>
              {indexToNumber(index)}. {question}
            </Typography>
            <Paragraph>{answer}</Paragraph>
          </Box>
        );
      })}
    </>
  );
}

export default Faq;
