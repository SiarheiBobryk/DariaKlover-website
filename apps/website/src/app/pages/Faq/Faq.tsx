import * as React from 'react';
import { Helmet } from 'react-helmet';

import FaqMDX from './faq.mdx';
import faqMetaData from './faqMetaData';

function Faq() {
  return (
    <>
      <Helmet>
        <title>{faqMetaData.title}</title>
      </Helmet>
      <FaqMDX />
    </>
  );
}

export default Faq;
