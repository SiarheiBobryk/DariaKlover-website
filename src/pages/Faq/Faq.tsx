import * as React from 'react';
import { Helmet } from 'react-helmet';

import Heading from '../../components/Heading';
import faqMetaData from './faqMetaData';

function Faq() {
  return (
    <>
      <Helmet>
        <title>{faqMetaData.title}</title>
      </Helmet>
      <Heading>{faqMetaData.heading}</Heading>
      {/* TODO: Make a separate components for every messager */}
    </>
  );
}

export default Faq;
