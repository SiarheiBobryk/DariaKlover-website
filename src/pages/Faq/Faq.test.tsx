import * as React from 'react';
import { BrowserRouter } from 'react-router-dom';

import { axe, toHaveNoViolations } from 'jest-axe';
import { render, RenderResult } from '@testing-library/react';
import renderer, { ReactTestRendererJSON } from 'react-test-renderer';

import Faq from './Faq';

expect.extend(toHaveNoViolations);

describe('<Faq />', () => {
  describe('a11y', () => {
    it('should match a11y standards', async () => {
      const { container }: RenderResult = render(
        <BrowserRouter>
          <Faq />
        </BrowserRouter>,
      );

      const results = await axe(container);
      expect(results).toHaveNoViolations();
    });
  });

  describe('snapshotting', () => {
    it('should render correctly', () => {
      const howToBook: ReactTestRendererJSON | ReactTestRendererJSON[] | null = renderer
        .create(
          <BrowserRouter>
            <Faq />
          </BrowserRouter>,
        )
        .toJSON();
      expect(howToBook).toMatchSnapshot();
    });
  });
});
