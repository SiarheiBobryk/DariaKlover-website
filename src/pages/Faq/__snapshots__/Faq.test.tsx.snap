// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Faq /> snapshotting should render correctly 1`] = `
Array [
  <h2
    className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
  >
    Часто задаваемые вопросы
  </h2>,
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardWarning MuiAlert-standard css-rob5aa-MuiPaper-root-MuiAlert-root"
    role="alert"
  >
    <div
      className="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
        data-testid="ReportProblemOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 5.99L19.53 19H4.47L12 5.99M12 2L1 21h22L12 2zm1 14h-2v2h2v-2zm0-6h-2v4h2v-4z"
        />
      </svg>
    </div>
    <div
      className="MuiAlert-message css-1pxa9xg-MuiAlert-message"
    >
      Страница находится в разработке
    </div>
  </div>,
  <ul
    className="MuiList-root MuiList-padding css-h4y409-MuiList-root"
  >
    <li
      className="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
      disabled={false}
    >
      <a
        className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
        href="#no-slots"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <h3
          className="MuiTypography-root MuiTypography-body1 css-p3gdy2-MuiTypography-root"
        >
          1
          . 
          Хочу записаться на консультацию, но нет подходящего места. Что делать?
        </h3>
      </a>
    </li>
    <li
      className="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
      disabled={false}
    >
      <a
        className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
        href="#how-to-prepare"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <h3
          className="MuiTypography-root MuiTypography-body1 css-p3gdy2-MuiTypography-root"
        >
          2
          . 
          Как подготовиться к сессии?
        </h3>
      </a>
    </li>
    <li
      className="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
      disabled={false}
    >
      <a
        className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
        href="#what-is-request"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <h3
          className="MuiTypography-root MuiTypography-body1 css-p3gdy2-MuiTypography-root"
        >
          3
          . 
          Что такое "запрос"?
        </h3>
      </a>
    </li>
    <li
      className="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
      disabled={false}
    >
      <a
        className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
        href="#age-restrictions"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <h3
          className="MuiTypography-root MuiTypography-body1 css-p3gdy2-MuiTypography-root"
        >
          4
          . 
          Со скольки лет можно работать с коучем?
        </h3>
      </a>
    </li>
    <li
      className="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
      disabled={false}
    >
      <a
        className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
        href="#sessions-amount-required"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <h3
          className="MuiTypography-root MuiTypography-body1 css-p3gdy2-MuiTypography-root"
        >
          5
          . 
          Сколько нужно сессий, чтобы решить запрос?
        </h3>
      </a>
    </li>
  </ul>,
  <section
    className="MuiBox-root css-1ebprri"
    id="no-slots"
  >
    <h3
      className="MuiTypography-root MuiTypography-body1 css-p3gdy2-MuiTypography-root"
    >
      1
      . 
      Хочу записаться на консультацию, но нет подходящего места. Что делать?
    </h3>
    <p
      className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-d3wcwz-MuiTypography-root"
    >
      Пожалуйста, свяжитесь со мной через 
      <a
        className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
        href="https://t.me/dariaklover"
        onBlur={[Function]}
        onFocus={[Function]}
        rel="noopener"
        target="_blank"
      >
        Telegram
         
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-1xzgpfl-MuiSvgIcon-root"
          data-testid="OpenInNewIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
          />
        </svg>
      </a>
      , 
      <a
        className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
        href="https://wa.me/+375447841051"
        onBlur={[Function]}
        onFocus={[Function]}
        rel="noopener"
        target="_blank"
      >
        WhatsApp
         
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-1xzgpfl-MuiSvgIcon-root"
          data-testid="OpenInNewIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
          />
        </svg>
      </a>
       или по
       
      <a
        className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
        href="mailto: dariaklover@gmail.com"
        onBlur={[Function]}
        onFocus={[Function]}
        rel="noopener"
        target="_blank"
      >
        Email (Электронная почта)
         
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-1xzgpfl-MuiSvgIcon-root"
          data-testid="OpenInNewIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
          />
        </svg>
      </a>
      .
    </p>
  </section>,
  <section
    className="MuiBox-root css-1ebprri"
    id="how-to-prepare"
  >
    <h3
      className="MuiTypography-root MuiTypography-body1 css-p3gdy2-MuiTypography-root"
    >
      2
      . 
      Как подготовиться к сессии?
    </h3>
    <p
      className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-d3wcwz-MuiTypography-root"
    >
      К сессии необходимо подготовить запрос. Тихое и комфортное место, где вас не будут отвлекать. Блокнот для записей и ручку.
    </p>
  </section>,
  <section
    className="MuiBox-root css-1ebprri"
    id="what-is-request"
  >
    <h3
      className="MuiTypography-root MuiTypography-body1 css-p3gdy2-MuiTypography-root"
    >
      3
      . 
      Что такое "запрос"?
    </h3>
    <p
      className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-d3wcwz-MuiTypography-root"
    >
      Запрос - это задача, которую вы хотите решить с коучем в ходе сессии.
    </p>
  </section>,
  <section
    className="MuiBox-root css-1ebprri"
    id="age-restrictions"
  >
    <h3
      className="MuiTypography-root MuiTypography-body1 css-p3gdy2-MuiTypography-root"
    >
      4
      . 
      Со скольки лет можно работать с коучем?
    </h3>
    <p
      className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-d3wcwz-MuiTypography-root"
    >
      С 18 лет.
    </p>
  </section>,
  <section
    className="MuiBox-root css-1ebprri"
    id="sessions-amount-required"
  >
    <h3
      className="MuiTypography-root MuiTypography-body1 css-p3gdy2-MuiTypography-root"
    >
      5
      . 
      Сколько нужно сессий, чтобы решить запрос?
    </h3>
    <p
      className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-d3wcwz-MuiTypography-root"
    >
      Все индивидуально. Зависит от сложности запроса, вашей вовлеченности и комфортного для вас темпа решения задачи.
    </p>
  </section>,
]
`;
